sorts/binary_insertion_sort.py

```diff
@@ -0,0 +1,93 @@
+"""
+A pure Python implementation of the binary insertion sort algorithm
+
+Binary insertion sort is a sorting algorithm which is similar to the insertion sort,
+but instead of using linear search to find the location where an element should be
+inserted, we use binary search. Thus, we reduce the comparative value of inserting
+a single element from O (N) to O (log N).
+
+For doctests run following command:
+python3 -m doctest -v binary_insertion_sort.py
+
+For manual testing run:
+python3 binary_insertion_sort.py
+"""
+
+
+def binary_search(collection: list, val: int, start: int, end: int) -> int:
+    """
+    A pure Python implementation of the binary search  algorithm.
+
+    :param collection: some ascending sorted collection with comparable items
```

Comment:
The docstring is a direct copy paste from `searches/binary_search.py`. Please update it as per the function.